src/Plugin.php

```diff
@@ -49,9 +49,7 @@ public static function getSubscribedEvents()
 
     public function onPreFileDownload(CPlugin\PreFileDownloadEvent $ev)
     {
-        $url = $ev->getProcessedUrl();
-
-        if (preg_match('/^https?/', $url)) {
+        if ('http' === substr($ev->getProcessedUrl(), 0, 4)) {
```

Comment:
I know but as you're leaving `s` not mandatory, you don't really care. You can't have an HTTP url with something else than `http` or `https` scheme. Maybe the other solution to avoid the `preg_match()` function call for such a simple test could be to use `stripos()` or `parse_url` like below:
```
$scheme = parse_url($url, PHP_URL_SCHEME);
if (null === $scheme || !in_array($scheme, ['http', 'https']) {
return;
}
```